python_modules/dagster/dagster/_core/definitions/asset_graph.py

```diff
@@ -165,6 +171,9 @@ def from_assets(
         code_versions_by_key: Dict[AssetKey, Optional[str]] = {}
         is_observable_by_key: Dict[AssetKey, bool] = {}
         auto_observe_interval_minutes_by_key: Dict[AssetKey, Optional[float]] = {}
+        required_assets_and_checks_by_key: Dict[
+            AssetKeyOrCheckKey, AbstractSet[AssetKeyOrCheckKey]
+        ] = defaultdict(set)
```

Comment:
Not a fan of using `defaultdict` here, since the type doesn't communicate that information. Either change the default or change the type. Otherwise, downstream consumers of this API will continue to be defensive when accessing this dict